packages/cli/src/telemetry/index.ts

```diff
@@ -0,0 +1,126 @@
+/// <reference lib="dom" />
+import fs from 'fs-extra';
+import uuid from 'uuid/v4';
+import path from 'path';
+import crypto from 'crypto';
+import envPaths from 'env-paths';
+import mapValues from 'lodash.mapvalues';
+import inquirer from 'inquirer';
+import proc from 'child_process';
+
+import { DISABLE_INTERACTIVITY } from '../prompts/prompt';
+import { Params } from '../scripts/interfaces';
+import ProjectFile from '../models/files/ProjectFile';
+
+type Field = string | number | boolean;
+
+interface GlobalTelemetryOptions {
+  optIn: boolean;
+  uuid: string;
+  salt: string;
+}
+
+export type CommandData = Params & {
+  name: string;
+  network?: string;
+};
+
+export default {
+  async report(commandName: string, options: Params, interactive: boolean): Promise<void> {
+    const telemetry = await checkOptIn(interactive);
+    if (telemetry === undefined || !telemetry.optIn) return;
+
+    // extract network name if present
+    let network;
+    if ('network' in options) {
+      network = options.network;
+      delete options.network;
+      if (network.match(/dev-/)) network = 'development';
+    }
+
+    // Conceal data before sending it
+    const concealedData = concealData(options, telemetry.salt);
+    const commandData: CommandData = { ...concealedData, name: commandName };
+    if (network !== undefined) commandData.network = network;
+
+    this.sendToFirebase(telemetry.uuid, commandData);
+  },
+
+  sendToFirebase(uuid: string, commandData: CommandData): void {
+    // We send to Firebase in a child process so that the CLI is not blocked from exiting.
+    const child = proc.fork(path.join(__dirname, './send-to-firebase'));
+    child.send({ uuid, commandData });
+
+    // Allow this process to exit while the child is still alive.
+    child.disconnect();
+    child.unref();
+  },
+};
+
+async function checkOptIn(interactive: boolean): Promise<GlobalTelemetryOptions | undefined> {
+  const project = new ProjectFile();
+  const localOptIn = project.telemetryOptIn;
+
+  const { data: globalDataDir } = envPaths('openzeppelin-sdk');
+  const globalDataPath = path.join(globalDataDir, 'telemetry.json');
+  let globalOptIn: GlobalTelemetryOptions | undefined = await fs.readJson(globalDataPath).catch(() => undefined);
+
+  if (localOptIn === false) return undefined;
+
+  // disable interactivity manually for tests and CI
+  if (DISABLE_INTERACTIVITY) interactive = false;
+
+  if (globalOptIn === undefined && interactive) {
+    const { telemetry } = await inquirer.prompt({
+      name: 'telemetry',
+      type: 'confirm',
+      message:
+        'Would you like to contribute anonymous usage data to help us improve the OpenZeppelin CLI? Learn more at https://zpl.in/telemetry',
```

Comment:
We should make sure this link is up before the release

Reply:
The link has been set up in zpl.in. The FAQ entry doesn't exist yet, though. It will exist once we put out the new release and publish that on the docs site.